zerver/lib/bugdown/__init__.py

```diff
@@ -961,6 +961,8 @@ def twitter_link(self, url: str) -> Optional[Element]:
 
     def get_url_data(self, e: Element) -> Optional[Tuple[str, str]]:
         if e.tag == "a":
+            if self.markdown.zulip_message is not None:
+                self.markdown.zulip_message.has_link = True
```

Comment:
While I think this might work, it's definitely not going to result in readable code.

I'd prefer that we do a preparatory refactor commit than modify this data-fetching function like this.  I'd like us to have a single code path for managing this sort of thing.  I believe the right way to do that is to (1) make sure we're actually calling `fixup_link` on all `a` tags we make for setting `target=_blank`, and then in a second commit,(2) Modify that function to take care of tracking this (and eventually `has_attachment` too).

Check out our [GitHub guide](https://zulip.readthedocs.io/en/latest/git-guide.html) and [commit guidelines](https://zulip.readthedocs.io/en/latest/contributing/version-control.html) for more details.